### PR TITLE
8324865: windows-x64-slowdebug still does not build after JDK-8324840

### PR DESCRIPTION
**Pull request by vnkozlov**
[~jwaters] commented in the PR [#17613](https://github.com/openjdk/jdk/pull/17613) that on windows we have strtok_s instead of strtok_r. We may need to #include "runtime/os.hpp" which defined strtok_r for windows instead of <string.hpp>. 

Including os.hpp will increase compilation time for all files which inclide "stringUtils.hpp".
I suggest to move code from .hpp to .cpp file. It is used only by c2 and inlining of this code is not too important.

Tested tier1 and tier2 builds with precompiled headers off.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324865](https://bugs.openjdk.org/browse/JDK-8324865): windows-x64-slowdebug still does not build after JDK-8324840 (**Bug** - P2)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17622/head:pull/17622` \
`$ git checkout pull/17622`

Update a local copy of the PR: \
`$ git checkout pull/17622` \
`$ git pull https://git.openjdk.org/jdk.git pull/17622/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17622`

View PR using the GUI difftool: \
`$ git pr show -t 17622`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17622.diff">https://git.openjdk.org/jdk/pull/17622.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17622#issuecomment-1915703607)

### CODE REVIEW

**Comment by vnkozlov** [comment]
Thank you @dholmes-ora. Yes, simplest solution is copy definition from os.hpp.

**Comment by vnkozlov** [comment]
@dholmes-ora, I used your suggestion and tested it (passed).

File: src/hotspot/share/utilities/stringUtils.hpp [hunk 1]
```diff
@@ -27,7 +27,10 @@
 
 #include "memory/allStatic.hpp"
 
-#include <string.h>
+#ifdef _WINDOWS
+  // strtok_s is the Windows thread-safe equivalent of POSIX strtok_r
```
Comment (dholmes-ora) [inline src/hotspot/share/utilities/stringUtils.hpp:31]: Nit: indent needs fixing

**Review by dholmes-ora — APPROVED** [review]
One nit else good.

Thanks

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up. Since this is a build fix, please integrate before 24 hours have passed.

File: src/hotspot/share/utilities/stringUtils.hpp (hunk 1 above)
Comment (dcubed-ojdk) [reply src/hotspot/share/utilities/stringUtils.hpp:31]: I'm not seeing the indent problem... Maybe my eyes are tired...

Comment (vnkozlov) [reply src/hotspot/share/utilities/stringUtils.hpp:31]: @dholmes-ora  Do you mean to remove 2 spaces before `//`?

Comment (vnkozlov) [reply src/hotspot/share/utilities/stringUtils.hpp:31]: This code is copy-pasted from os.hpp.

Comment (vnkozlov) [reply src/hotspot/share/utilities/stringUtils.hpp:31]: I am pushing it as it is since I got 2 reviews.

**Pull request closed without merging**